src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs

```diff
@@ -176,6 +173,28 @@ internal static Type TypeOfDateTimeOffsetAdapter
             }
         }
 
+        private static Type? s_typeOfMemoryStream;
+        internal static Type TypeOfMemoryStream
+        {
+            get
+            {
+                if (s_typeOfMemoryStream == null)
+                    s_typeOfMemoryStream = typeof(MemoryStream);
+                return s_typeOfMemoryStream;
+            }
+        }
```

Comment:
I realize you're just following the pattern used everywhere else, so keep what you have.  But in the future we might consider consolidating all of these to use null coalescing assignment, e.g.
```C#
internal static Type TypeOfMemoryStream => s_typeOfMemoryStream ??= typeof(MemoryStream);
```

Reply:
Honestly I think these statics could for the most part be eliminated. Seems like the original use case was to make type comparisons cheaper, but the JIT now optimizes `if (obj.GetType() == typeof(Foo))` to be very efficient. Might even give a perf gain if we were to refactor these.